From 64ff961b4c3e05261f9517089b6a5aa351799739 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Tue, 21 Jan 2025 16:36:42 +0100
Subject: [PATCH 1573/1695] dts: stm32: pass dummy parameters for EXTI phandle

The driver EXTI registers itself as type DT_DRIVER_INTERRUPT, so
the framework uses EXTI node property #interrupt-cells to find the
number of DT parameters that are required and passes them to the
driver callback on dt_driver_device_from_node_idx_prop().

Temporarily add two dummy parameters to prevent the framework to
access two unrelated data in the flatten DT.
Set the first dummy parameter to a magic unique value 0xffff. This
will make easy to detect it.
These wakeup-parent properties will be dropped soon.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 80b98b94641472852046b7903f884462eaff7b97 dts: stm32: pass dummy parameters for EXTI phandle]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I749ba91143b5ee56d50c86ebde4b00611bfe50b6
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434920
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi     | 2 +-
 core/arch/arm/dts/stm32mp135f-dk.dts  | 2 +-
 core/arch/arm/dts/stm32mp211.dtsi     | 2 +-
 core/arch/arm/dts/stm32mp231.dtsi     | 4 ++--
 core/arch/arm/dts/stm32mp235f-dk.dts  | 2 +-
 core/arch/arm/dts/stm32mp251.dtsi     | 4 ++--
 core/arch/arm/dts/stm32mp257f-dk.dts  | 2 +-
 core/arch/arm/dts/stm32mp257f-ev1.dts | 2 +-
 8 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index 0fdc0ceed..b94906ad9 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -293,7 +293,7 @@
 
 		pwr_irq: pwr@50001010 {
 			compatible = "st,stm32mp1-pwr-irq";
-			wakeup-parent = <&exti>;
+			wakeup-parent = <&exti 0xffff 0>;
 			wakeup-gpios = <0>, <0>, <0>, <0>, <0>, <0>;
 			status = "disabled";
 		};
diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index 155afd351..f15b93a6d 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -686,7 +686,7 @@
 	st,tamp-passive-nb-sample = <4>;
 	st,tamp-passive-sample-clk-div = <16384>;
 	wakeup-source;
-	wakeup-parent = <&exti>;
+	wakeup-parent = <&exti 0xffff 0>;
 
 	/* Tamper button */
 	tamp-button {
diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index b0608d878..b0989ebde 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -763,7 +763,7 @@
 		pwr: pwr@44210000 {
 			compatible = "st,stm32mp21-pwr";
 			reg = <0x44210000 0x400>;
-			wakeup-parent = <&exti1>;
+			wakeup-parent = <&exti1 0xffff 0>;
 			interrupts = <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>;
 			wakeup-gpios = <0>, <0>, <0>, <0>, <0>, <0>;
 			status = "disabled";
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 836eacd8a..772089331 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -827,7 +827,7 @@
 		pwr: pwr@44210000 {
 			compatible = "st,stm32mp25-pwr";
 			reg = <0x44210000 0x400>;
-			wakeup-parent = <&exti1>;
+			wakeup-parent = <&exti1 0xffff 0>;
 			interrupts = <GIC_SPI 226 IRQ_TYPE_LEVEL_HIGH>;
 			wakeup-gpios = <0>, <0>, <0>, <0>, <0>, <0>;
 			status = "disabled";
@@ -1130,7 +1130,7 @@
 			clocks = <&rcc CK_BUS_RTC>, <&rcc RTC_CK>;
 			clock-names = "pclk", "rtc_ck";
 			wakeup-source;
-			wakeup-parent = <&exti2>;
+			wakeup-parent = <&exti2 0xffff 0>;
 			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
diff --git a/core/arch/arm/dts/stm32mp235f-dk.dts b/core/arch/arm/dts/stm32mp235f-dk.dts
index 91c2054f1..0fc1c67f5 100644
--- a/core/arch/arm/dts/stm32mp235f-dk.dts
+++ b/core/arch/arm/dts/stm32mp235f-dk.dts
@@ -454,7 +454,7 @@
 
 &tamp {
 	wakeup-source;
-	wakeup-parent = <&exti2>;
+	wakeup-parent = <&exti2 0xffff 0>;
 	st,tamp-passive-precharge = <2>;
 	st,tamp-passive-nb-sample = <4>;
 	st,tamp-passive-sample-clk-div = <16384>;
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 71b09cd68..44340a4f4 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -914,7 +914,7 @@
 		pwr: pwr@44210000 {
 			compatible = "st,stm32mp25-pwr";
 			reg = <0x44210000 0x400>;
-			wakeup-parent = <&exti1>;
+			wakeup-parent = <&exti1 0xffff 0>;
 			interrupts = <GIC_SPI 226 IRQ_TYPE_LEVEL_HIGH>;
 			wakeup-gpios = <0>, <0>, <0>, <0>, <0>, <0>;
 			status = "disabled";
@@ -1217,7 +1217,7 @@
 			clocks = <&rcc CK_BUS_RTC>, <&rcc RTC_CK>;
 			clock-names = "pclk", "rtc_ck";
 			wakeup-source;
-			wakeup-parent = <&exti2>;
+			wakeup-parent = <&exti2 0xffff 0>;
 			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index c7d2563b3..b90d6f7e5 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -458,7 +458,7 @@
 
 &tamp {
 	wakeup-source;
-	wakeup-parent = <&exti2>;
+	wakeup-parent = <&exti2 0xffff 0>;
 	st,tamp-passive-precharge = <2>;
 	st,tamp-passive-nb-sample = <4>;
 	st,tamp-passive-sample-clk-div = <16384>;
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index dc302f7df..860b30c02 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -490,7 +490,7 @@
 
 &tamp {
 	wakeup-source;
-	wakeup-parent = <&exti2>;
+	wakeup-parent = <&exti2 0xffff 0>;
 	st,tamp-passive-precharge = <2>;
 	st,tamp-passive-nb-sample = <4>;
 	st,tamp-passive-sample-clk-div = <16384>;
-- 
2.47.2

